### PR TITLE
[Engine][Task] Add cancel Task

### PR DESCRIPTION
**Pull request by ic4y**
Add task cancel and modify the test code

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by EricJoy2048 — APPROVED** [review]
+1

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java [hunk 1]
```diff
@@ -87,16 +87,15 @@ public void testCancel() {
...
 
-        CompletableFuture<TaskExecutionState> completableFuture = taskExecutionService.submitTaskGroup(new TaskGroupDefaultImpl("ts", Lists.newArrayList(testTask1,testTask2)), cancellationFuture);
+        long taskGroupId = flakeIdGenerator.newId();
+        CompletableFuture<TaskExecutionState> completableFuture = taskExecutionService.deployLocalTask(new TaskGroupDefaultImpl(taskGroupId, "ts", Lists.newArrayList(testTask1, testTask2)), new CompletableFuture<>());
 
-        cancellationFuture.cancel(true);
+        taskExecutionService.cancelTaskGroup(taskGroupId);
 
-        await().atMost(sleepTime + 300, TimeUnit.MILLISECONDS).untilAsserted(()->{
+        await().atMost(sleepTime + 300, TimeUnit.MILLISECONDS).untilAsserted(() -> {
```
Comment (ashulin) [inline seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java]: Can't `CheckStyle` check for this style error?

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java [hunk 2]
```diff
@@ -129,10 +131,21 @@ public NonCompletableFuture<TaskExecutionState> deployTask(
 
             // TODO Use classloader load the connector jars and deserialize Task
             taskGroup = nodeEngine.getSerializationService().toObject(taskImmutableInfo.getGroup());
+            return deployLocalTask(taskGroup, resultFuture);
+        } catch (Throwable t) {
+            logger.severe(ExceptionUtils.getMessage(t));
```
Comment (ashulin) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java]: Can we output logs based on job level?

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java (hunk 1 above)
Comment (ic4y) [reply seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java]: My mistake, my idea CheckStyle didn't cover the test, I've fixed it

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java (hunk 2 above)
Comment (ic4y) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/TaskExecutionService.java]: This is an optimization point. Let's do it later.  Now temporarily add the JobID to the log

**Comment by ic4y** [comment]
@Hisoka-X PTAL Thanks

**Pull request merged**